### DISCUSSION TITLE
Any option / feature to preload the next and previous image on lazyload

### DISCUSSION THREAD

**Discussion by asdutoit** [davidjerleke/embla-carousel#306] [Ideas]
I am not sure if this solution is already available, but I could not find it in the documentation. 

With `lazyloading` enabled, an option to include a prop that specifies that the next and previous images are pre-loaded would be greatly advantageous.  This would lead to a great user experience. 
Airbnb for example, Implements this in their carousels.

**Comment by davidjerleke** [comment]
Hi @asdutoit,

Thank you for your question. At the time of writing, the only example of basic **lazy loading** is [this CodeSandbox](https://www.embla-carousel.com/examples/inspiration/#lazy-load). Unfortunately it doesn't support what you want to do. When I created the CodeSandbox, Embla didn't have its plugin system, which was later [released with v6](https://github.com/davidcetinkaya/embla-carousel/releases/tag/v6.0.0). I think it would be nice to create a `embla-carousel-lazy-load` [plugin](https://www.embla-carousel.com/plugins/), but don't know when and if I get the time to do it.

I've added Lazy Load to the plugin suggestion list:

- #284

Best,
David

**Reply by asdutoit** [reply]
Hi @davidjerleke ,

No problem at all and thanks for getting back to me. 
I see Embla does get a lot of love online and I concur, it is a fantastic library. 

Appreciate you adding the feature to your list of considerations for plugin development.

**Reply by davidjerleke** [reply]
Thanks @asdutoit. I just want to mention that you can probably achieve what you want with the Embla API, although not as convenient as adding a plugin of course. If I get the chance to work on a lazy load plugin I’ll ask you to join the conversation about what features it should have.

In the meantime, be sure to check out the upcoming **version 7** of Embla:

- #298 

Best,
David

**Reply by asdutoit** [reply]
Will do!  Thanks

**Comment by MaxDesignFR** [comment]
From my experience and network testing on chrome, using native [`loading="lazy"`]((https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading)) works well. Browser support is not perfect yet but decent, it is a neat and easy addition to have image lazyload progressively as the carrousel is sliding.

**Comment by davidjerleke** [comment]
Closing as this is either considered a DIY thing with the library, or could be achieved by creating a plugin. This is not something that will be built into the core.
